### PR TITLE
update security to implement recipient id

### PR DESCRIPTION
**Pull request by WAvdBeek**

### CODE REVIEW

File: api/oc_knx_sec.c [hunk 1]
```diff
@@ -1237,30 +1233,30 @@ oc_at_load_entry(int entry)
...
                           oc_string(rep->value.string),
                           oc_string_len(rep->value.string));
           }
+          if (rep->iname == 847) {
+            oc_free_string(&g_at_entries[entry].osc_rid);
+            oc_new_string(&g_at_entries[entry].osc_rid,
+                          oc_string(rep->value.string),
+                          oc_string_len(rep->value.string));
+          }
+          if (rep->iname == 840) {
```
Comment (tiniuclx) [inline api/oc_knx_sec.c:1254]: Do you think we should make an enum for these magic numbers? E.g. OSC_CONTEXTID_KEY = 846, OSC_RID_KEY = 847...

File: security/oc_oscore_context.c [hunk 2]
```diff
@@ -196,93 +198,126 @@ oc_oscore_free_context(oc_oscore_context_t *ctx)
...
+  PRINT("  desc      : %s\n", desc);
+  PRINT("  index     : %d\n", auth_at_index);
+  PRINT("  sid size  : %d ", senderid_size);
+  oc_char_print_hex(senderid, senderid_size);
+  PRINT("\n");
+  PRINT("  rid size  : %d ", recipientid_size);
+  oc_char_print_hex(recipientid, recipientid_size);
+  PRINT("\n");
+  PRINT("  ctx size  : %d\n", osc_ctx_size);
+  PRINT("  ms size   : %d\ ", mastersecret_size);
```
Comment (tiniuclx) [inline security/oc_oscore_context.c:252]: No `\n` here?

**Review by tiniuclx — APPROVED** [review]
Seems OK so far, you've been pretty thorough! A couple of minor things...

File: api/oc_knx_sec.c (hunk 1 above)
Comment (WAvdBeek) [reply api/oc_knx_sec.c:1256]: added info how it is build up

File: security/oc_oscore_context.c (hunk 2 above)
Comment (WAvdBeek) [reply security/oc_oscore_context.c]: fixed

File: api/oc_knx_client.c [hunk 3]
```diff
@@ -334,6 +399,8 @@ oc_knx_client_do_broker_request(char *resource_url, uint32_t ia,
                                 char *destination, char *rp)
 {
   char query[20];
+
+  // TODO: do the new discovery here
   snprintf(query, 20, "if=urn:knx:ia.%d", ia);
```
[the remark this answers is not in the document]
Comment (WAvdBeek) [reply api/oc_knx_client.c:404]: new query params

File: api/oc_knx_client.c [hunk 4]
```diff
@@ -254,11 +259,60 @@ do_credential_exchange(oc_client_response_t *data)
...
+}
+
 int
-oc_initiate_spake(oc_endpoint_t *endpoint, char *password, char *oscore_id)
+oc_initiate_spake(oc_endpoint_t *endpoint, char *password, char *recipient_id)
 {
   int return_value = -1;
 
+  // sort this one out later..
+  return return_value;
```
[the remark this answers is not in the document]
Comment (WAvdBeek) [reply api/oc_knx_client.c:313]: remove comment

**Comment by tiniuclx** [comment]
~format

**Comment by tiniuclx** [comment]
~format

**Pull request merged**